controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -58,7 +73,8 @@ type WorkloadCluster interface {
 	EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 
 	// Behaviors necessary for upgrade
-
+	GetCoreDNSInfo(ctx context.Context) (string, string, *appsv1.Deployment, error)
```

Comment:
Not blocking: this can be grouped under "Basic health and status command" above.